p8e-util/src/main/kotlin/io/p8e/crypto/Pen.kt

```diff
@@ -20,40 +20,48 @@ class Pen(privateKey: PrivateKey, publicKey: PublicKey) {
         val PROVIDER = "BC"
     }
 
-    val privateKey: PrivateKey
-    val lens: Lens
     init {
         Security.addProvider(BouncyCastleProvider())
-        this.privateKey = privateKey
-        lens = Lens(publicKey)
+    }
+
+    var keys: KeyPair? = null
+    var lens: Lens? = null
+
+    override fun setKeyId(keyPair: KeyPair) {
+        keys = keyPair
+        lens = Lens(keys!!.public)
+    }
+
+    override fun setKeyId(uuid: String) {
+        /*no-op*/
```

Comment:
I wonder if it would be worth just populating the uuid on the affiliate table even when using the default driver and just fetch the keypair from the db in here so we could always use a uuid externally?

Reply:
Yep, I have a ticket to remove passing keyPairs around and instead each keypair is referenced by a UUID and we can pull it that way.

Reply:
It is maybe worth considering that instead of a uuid the address of the key should be used instead because unlike a uuid it is directly computed from the key... also everything blockchain based will use an address... ¯\_(ツ)_/¯

Reply:
Aren't you on vacation @iramiller? 😄

Reply:
Agreed, something worth considering now, this might require some conversion effort of keys within p8e cause not sure it makes sense to store the actual key + the address of that key. 

_**I shouldn't answer reply to someone that is vacationing wrong**_

Reply:
do we generate this uuid, or does that just come straight from SmartKey when we generate a key? I.e. we might not even control this?

Reply:
they provide us a UUID for a SecurityObject (a locker for keys) is my real world analogy.